software/bwa/mem/main.nf

```diff
@@ -32,13 +31,15 @@ process BWA_MEM {
     def prefix     = options.suffix ? "${meta.id}${options.suffix}" : "${meta.id}"
     def read_group = meta.read_group ? "-R ${meta.read_group}" : ""
     """
+    INDEX=`find -L ./ -name "*.amb" | sed 's/.amb//'`
```

Comment:
I still don't think this is the right way to do it...
And I think you're missing some indices as well

Reply:
This is just getting the genome prefix automatically. You don't need to list all of the files in the index for that. Just one.

Reply:
why don't you keep the fasta as an input?

Reply:
This also means you don't need to stage the `fasta` file in the module too and pipelines can accept tar archives / untarred directores containing the indices. You don't even need to know what all of those indices file extensions are as long as the directory contains the index. Much cleaner.

Reply:
yes, but if you don't have the fasta as an input, how will it work?
Also don't you need the fasta index as well?
or do you include that in the index path?

Reply:
I think the easiest option is to try that command in a directory containing a BWA index and see what it returns 😉